brain/circuit.cpp

```diff
@@ -103,10 +103,20 @@ class Circuit::Impl
     {}
     virtual ~Impl() {}
 
+    virtual size_t getNumNeurons() const = 0;
+
     GIDSet getGIDs( const std::string& target ) const
     {
-        return brion::Target::parse(
-            _targetParsers, target.empty() ? _circuitTarget : target );
+        if( target.empty( ))
+        {
+            GIDSet gids;
+            for( size_t i = 0; i < getNumNeurons(); ++i )
+                gids.insert( i + 1 );
+            return gids;
+        }
+        else
```

Comment:
We had a discussion this morning and agreed with that solution, but I guess you should have been in the loop. It seemed ok to assume that specifying no target would mean getting all the GIDs. What you are proposing as an alternative was also mentioned during the discussion and can clearly be done without modifying Brion. To give you a bit of background, the idea is that we must be able to open and visualize a circuit even if it does not contain any target yet. From what you are saying, modifying Brion is the wrong idea?